aws/lambda/tests/test_lambda_oss_ci_job_queue_time.py

```diff
@@ -0,0 +1,263 @@
+                1742262372,
+            ),
+        ]
+
+    return [
+        (
+            60000,
+            "pytorch/pytorch",
+            "inductor-h100",
+            "test1 (h100, 5, 5, linux.aws.h100)",
+            "runs/1/job/1",
+            "linux.aws.h100",
+            1742262372,
+        ),
+        (
+            50000,
+            "pytorch/pytorch",
+            "inductor-h100",
+            "test1 (h100, 5, 5, linux.aws.h100)",
+            "runs/1/job/2",
+            "linux.aws.h100",
+            1742262372,
+        ),
+        (
+            55000,
+            "pytorch/pytorch",
+            "inductor-h100",
+            "test1 (h100, 2, 6, linux.aws.h100)",
+            "runs/1/job/3",
+            "linux.aws.h100",
+            1742262372,
+        ),
+        (
+            1729,
+            "pytorch/pytorch",
+            "inductor-h100",
+            "test2 (h100, 1, 1, linux.aws.h100)",
+            "runs/2/job/1",
+            "linux.aws.h100",
+            1742262372,
+        ),
+        (
+            1352,
+            "pytorch/pytorch",
+            "inductor-rocm",
+            "rocm-test1(1, 1, linux.rocm.gpu.2)",
+            "runs/3/job/1",
+            "linux.rocm.gpu.2",
+            1742262372,
+        ),
+        (
+            1400,
+            "pytorch/pytorch",
+            "inductor-rocm",
+            "rocm-test1 (1, 1, linux.rocm.gpu.2)",
+            "runs/4/job/2",
+            "linux.rocm.gpu.2",
+            1742262372,
+        ),
+    ]
+
+
+def get_default_result_columns() -> Tuple:
+    return (
+        "queue_s",
+        "repo",
+        "workflow_name",
+        "job_name",
+        "html_url",
+        "machine_type",
+        "time",
+    )
+
+
+def mock_query_result(
+    query: str, parameters: str, rows_in_queue: List[Tuple], rows_picked: List[Tuple]
+) -> Any:
+    result = MagicMock()
+    if "LENGTH(job.steps) = 0" in query:
+        result.column_names = get_default_result_columns()
+        result.result_rows = rows_in_queue
+    if "LENGTH(job.steps) != 0'" in query:
+        result.column_names = get_default_result_columns()
+        result.result_rows = rows_picked
+    return result
+
+
+def mock_s3_resource_put(mock_s3_resource: Any) -> None:
+    mock_s3 = mock_s3_resource.return_value
+    mock_object = mock_s3.Object.return_value
+    mock_object.put.return_value = {"ResponseMetadata": {"HTTPStatusCode": 200}}
+
+
+def get_mock_s3_resource_object(mock_s3_resource: Any):
+    return mock_s3_resource.return_value.Object
+
+
+def mock_db_client(
+    mock: Any,
+    rows_in_queue: List[Tuple] = get_default_result_rows(),
+    rows_picked: List[Tuple] = [],
+) -> None:
+    mock_client = mock.return_value
+    mock_client.query.side_effect = lambda query, parameters: mock_query_result(
+        query, parameters, rows_in_queue, rows_picked
+    )
+
+
+def set_default_env_variables():
+    os.environ["CLICKHOUSE_ENDPOINT"] = "https://clickhouse.test1"
```

Comment:
it is a terrible engineering standard overwrite the exported environ and hide its original value.

Better use the environ as your default for your cli parameters, than the opposite